taxi/views.py

```diff
@@ -1,47 +1,61 @@
+from django.contrib.auth.decorators import login_required
+from django.contrib.auth.mixins import LoginRequiredMixin
 from django.shortcuts import render
-from django.views import generic
+from django.views.generic import ListView, DetailView
+from django.http.request import HttpRequest
+from django.http.response import HttpResponse
 
-from .models import Driver, Car, Manufacturer
+from taxi.models import Driver, Car, Manufacturer
 
 
-def index(request):
+@login_required
+def index(request: HttpRequest) -> HttpResponse:
     """View function for the home page of the site."""
+    if not request.session.get("num_visits"):
+        request.session["num_visits"] = 0
 
     num_drivers = Driver.objects.count()
     num_cars = Car.objects.count()
     num_manufacturers = Manufacturer.objects.count()
-
+    request.session["num_visits"] = request.session.get("num_visits") + 1
     context = {
         "num_drivers": num_drivers,
         "num_cars": num_cars,
         "num_manufacturers": num_manufacturers,
+        "num_visits": request.session.get("num_visits"),
+        "f": request.session.items()
```

Comment:
`f` bad name

Reply:
Wow-wow.. I forgot about that testing key... my bad.